src/providers/object-provider.js

```diff
@@ -300,68 +300,121 @@ export default class YamcsObjectProvider {
-                        source: 'timestamp',
-                        name: 'Timestamp',
-                        format: 'iso',
-                        hints: {
-                            domain: 1
-                        }
-                    }
-                ]
-            };
-
-            if (obj.type === OBJECT_TYPES.STRING_OBJECT_TYPE) {
-                obj.telemetry.values[0].hints = {};
-            } else if (obj.type === OBJECT_TYPES.IMAGE_OBJECT_TYPE) {
-                obj.telemetry.values[0].hints = { image: 1 };
-                obj.telemetry.values[0].format = 'image';
+            aggregateHasMembers = this.aggregateHasMembers(parameter);
+            obj.composition = [];
+            if (aggregateHasMembers) {
+                let memberMetadata = this.formatAggregateMembers(parameter.type.member);
+                obj.telemetry.values = obj.telemetry.values.concat(memberMetadata);
             }
         }
 
         this.addObject(obj);
 
         parent.composition.push(obj.identifier);
 
-        if (isAggregate) {
-            if (parameter.type.member !== undefined) {
-                parameter.type.member.forEach(member => {
-                    let memberQualifiedName = qualifiedName + '.' + member.name;
-                    /* Use current name as a prefix for the member name. */
-                    this.addParameter(member, memberQualifiedName, obj, name + '_');
+        if (aggregateHasMembers) {
+            parameter.type.member.forEach(member => {
+                let memberQualifiedName = qualifiedName + '.' + member.name;
+                /* Use current name as a prefix for the member name. */
+                this.addParameter(member, memberQualifiedName, obj, name + '_');
+            });
+        }
+    }
+
+    addHints(key, obj) {
+        let metadatum = obj.telemetry.values.find(md => md.key === key);
+
+        if (obj.type === OBJECT_TYPES.STRING_OBJECT_TYPE) {
+            metadatum.hints = {};
+        } else if (obj.type === OBJECT_TYPES.IMAGE_OBJECT_TYPE) {
+            metadatum.hints = { image: 1 };
+            metadatum.format = 'image';
+        }
+
+    }
+
+    isAggregate(parameter) {
+        let isAggregate = false;
+
+        if (parameter.type !== undefined) {
+            isAggregate = parameter.type.engType === 'aggregate';
+        }
+
+        return isAggregate;
+    }
+
+    formatAggregateMembers(members, parentKey = '', rangeHint = 1) {
+        let formatted = [];
+
+        members.forEach(member => {
+            let key = member.name;
+            let name = member.name;
+
+            if (parentKey) {
+                key = parentKey + '.' + key;
+                name = parentKey.split('.').shift() + ' ' + name;
+            }
+
+            if (!this.isAggregate(member)) {
+                formatted.push({
+                    key,
+                    name,
+                    hints: {
+                        range: rangeHint++
```

Comment:
Can we really assume that each member of an aggregate is a range? I think we need something more from yamcs telling us if the member is a domain or a range, no?

Reply:
yeah, I don't think assigning each one an incrementing range is correct... I'm not sure how or if we assign them hints

Reply:
This is fine as per our conversation with Andrew. Let's keep it.